### PR TITLE
Docs/fix trainer fct notebooks

### LINKED ISSUE #1008: Docu fix: `result = trainer.run()`
### Issue Description

In the provided code examples, it appears that the training function is not being used properly. Specifically, in both cases cited by the code-line links, the variable named `result` is being assigned a trainer class which is never used.

### Proposed Solution

To address this issue, I suggest changing the class instance from `result` to `trainer` and assign the ouput of its function `run()` to the variable `result`; using it as follows:

```python
# Before
result = OnpolicyTrainer()
print(result)

# After
trainer = OnpolicyTrainer()
result = trainer.run()
print(result)

```

or more efficiently:

```python
result = OnpolicyTrainer().run()
print(result)
```

### Affected Code Lines

https://github.com/thu-ml/tianshou/blob/ea48cc298905ad37fa67bd05b76f55079e1905c2/docs/02_notebooks/L6_Trainer.ipynb#L193-L204

https://github.com/thu-ml/tianshou/blob/ea48cc298905ad37fa67bd05b76f55079e1905c2/docs/02_notebooks/L7_Experiment.ipynb#L241-L252

### PR DESCRIPTION
**Pull request by carlocagnetta**
- [x] I have added the correct label(s) to this Pull Request or linked the relevant issue(s)
- [x] I have provided a description of the changes in this Pull Request
- [x] I have added documentation for my changes
- [x] If applicable, I have added tests to cover my changes.
- [x] I have reformatted the code using `poe format` 
- [x] I have checked style and types with `poe lint` and `poe type-check`
- [ ] (Optional) I ran tests locally with `poe test` 
(or a subset of them with `poe test-reduced`) ,and they pass
- [x] (Optional) I have tested that documentation builds correctly with `poe doc-build`

This PR resolves #1008

### CODE REVIEW

**Comment by carlocagnetta** [comment]
@opcode81 This is the PR for the issue we discussed today
@MischaPanch Could you please review it?
I also changed a minor thing in the `.rst` files of the docu to upgrade to `CarPole-v1`

**Comment by MischaPanch** [comment]
The build fails because you committed a different python version to the notebook metadata. Could you undo this pls?

**Comment by MischaPanch** [comment]
The build on windows fails, but it has nothing to do with the changes here. Seems some caching problem in CI. If it appears on master, I'm gonna solve it in a separate PR. Merging this, thanks for the docs fixes!

**Review by MischaPanch — APPROVED** [review]
